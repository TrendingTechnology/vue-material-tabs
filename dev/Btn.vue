<template>
  <button class="btn" @click="$emit('click')">
    <slot />
  </button>
</template>

<style scoped>
.btn {
  border: none;
  align-items: center;
  border-radius: 4px;
  display: inline-flex;
  flex: 0 0 auto;
  font-weight: 500;
  letter-spacing: 0.0892857143em;
  justify-content: center;
  outline: 0;
  position: relative;
  text-decoration: none;
  text-indent: 0.0892857143em;
  text-transform: uppercase;
  vertical-align: middle;
  white-space: nowrap;
  background-color: #1867c0;
  cursor: pointer;
  padding: 0.7rem 2rem;
  color: white;
  transition: background 0.28s;
}

.btn:hover {
  background: #1e7be6;
}
</style>
