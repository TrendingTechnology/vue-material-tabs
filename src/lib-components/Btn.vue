<template>
  <button class="btn" @click="$emit('click')">
    <svg
      xmlns="http://www.w3.org/2000/svg"
      viewBox="0 0 24 24"
      role="img"
      aria-hidden="true"
    >
      <path d="M15.41,16.58L10.83,12L15.41,7.41L14,6L8,12L14,18L15.41,16.58Z" />
    </svg>
  </button>
</template>

<script>
export default {
  name: "Btn",
};
</script>

<style scoped>
.btn {
  outline: none;
  border: none;
  background: transparent;
  cursor: pointer;
}

.btn svg {
  height: 24px;
  width: 24px;
}

.btn:disabled svg {
  fill: rgb(214, 213, 213);
  cursor: default;
}
</style>
