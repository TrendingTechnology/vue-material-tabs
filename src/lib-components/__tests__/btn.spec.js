import { mount } from "@vue/test-utils";
import Btn from "../Btn.vue";

describe("Btn.vue", () => {
  const mountFunction = (options = {}) => mount(Btn, options);

  it("Should mount he component", () => {
    const wrapper = mountFunction();

    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.vm).toBeDefined();
  });

  it("Should emit event 'click' when clicked ", async () => {
    const wrapper = mountFunction();

    await wrapper.trigger("click");

    expect(wrapper.emitted("click")).toBeTruthy();
    expect(wrapper.emitted("click").length).toBe(1);
  });
});
