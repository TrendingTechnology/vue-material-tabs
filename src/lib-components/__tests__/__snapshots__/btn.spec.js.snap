// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Btn.vue Should mount he component 1`] = `
"<button class=\\"btn\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 24 24\\" role=\\"img\\" aria-hidden=\\"true\\">
    <path d=\\"M15.41,16.58L10.83,12L15.41,7.41L14,6L8,12L14,18L15.41,16.58Z\\"></path>
  </svg></button>"
`;
